src/Language/DifferentialDatalog/Validate.hs

```diff
@@ -303,15 +306,21 @@ indexValidate d idx = do
     return idx'
 
 ruleValidate :: (MonadError String me, ?cfg::Config) => DatalogProgram -> Rule -> me Rule
-ruleValidate d rl@Rule{..} = do
-    when (not $ null ruleRHS) $ do
-        case head ruleRHS of
-             RHSLiteral True _ -> return ()
-             _                 -> err d (pos rl) "Rule must start with a positive literal"
-    mapIdxM_ (ruleRHSValidate d rl) ruleRHS
-    mapM_ (ruleLHSValidate d rl) ruleLHS
+ruleValidate d rl = do
```

Comment:
One would expect validation not to modify the program.
We should have a clearer staging of program rewriting.